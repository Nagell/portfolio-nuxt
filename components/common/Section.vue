<template>
    <section
        :id="headingId"
        :class="cn(props.class, 'py-16')"
    >
        <div class="container">
            <CommonTypography
                v-if="props.heading"
                tag="h2"
            >
                {{ heading }}
            </CommonTypography>
            <slot />
        </div>
    </section>
</template>

<script lang="ts" setup>

    import type { HTMLAttributes } from 'vue'

    interface Props {
        class?: HTMLAttributes['class']
        heading?: string
        id?: string
    }

    const props = defineProps<Props>()

    const headingId = computed(() => props.id || props.heading?.toLowerCase().replace(/\s/g, '-'))
</script>
