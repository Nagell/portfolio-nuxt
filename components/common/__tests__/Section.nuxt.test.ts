import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'

import Section from '../Section.vue'

describe('Section Component', async () => {
    it('renders correctly', async () => {
        const wrapper = await mountSuspended(Section, {
            slots: {
                default: '<div class="test-content">Test Content</div>'
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders slot content', async () => {
        const wrapper = await mountSuspended(Section, {
            slots: {
                default: '<div class="test-content">Test Content</div>'
            }
        })
        expect(wrapper.find('.test-content').exists()).toBe(true)
        expect(wrapper.find('.test-content').text()).toBe('Test Content')
    })

    it('applies custom class', async () => {
        const customClass = 'custom-section-class'
        const wrapper = await mountSuspended(Section, {
            props: { class: customClass }
        })
        expect(wrapper.classes()).toContain(customClass)
    })

    it('renders heading when provided', async () => {
        const heading = 'Test Heading'
        const wrapper = await mountSuspended(Section, {
            props: { heading }
        })
        expect(wrapper.find('h1').exists()).toBe(true)
        expect(wrapper.find('h1').text()).toBe(heading)
    })
})
