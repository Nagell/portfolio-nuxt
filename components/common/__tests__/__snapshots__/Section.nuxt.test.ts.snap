// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Section Component > renders correctly 1`] = `
"<section id="" class="py-16">
  <div class="container">
    <!--v-if-->
    <div class="test-content">Test Content</div>
  </div>
</section>"
`;
